src/container/srv_container.c

```diff
@@ -1938,6 +2058,22 @@ cont_open(struct rdb_tx *tx, struct ds_pool_hdl *pool_hdl, struct cont *cont,
 		D_GOTO(out, rc);
 	}
 
+	if ((in->coi_flags & DAOS_COO_EVICT_ALL) && !is_owner) {
+		D_ERROR(DF_CONT": only owner can evict all handles\n",
+			DP_CONT(cont->c_svc->cs_pool_uuid, cont->c_uuid));
+		daos_prop_free(prop);
+		rc = -DER_NO_PERM;
+		goto out;
+	}
+
+	if ((in->coi_flags & DAOS_COO_EX) && !is_owner) {
+		D_ERROR(DF_CONT": only owner can open exclusively\n",
+			DP_CONT(cont->c_svc->cs_pool_uuid, cont->c_uuid));
+		daos_prop_free(prop);
+		rc = -DER_NO_PERM;
+		goto out;
+	}
```

Comment:
Hm, I find myself wondering if maybe there is a middle road. I do like evict being a capability whose details are encapsulated in the security module. Conceptually it seems to fit the idea of a capability. In ds_sec_cont_get_capabilities, perhaps we could just do the ownership check and add this capability at the end, rather than doing the owner_min_capas approach? Since for the time being, containers are the only ones using these capabilities.

Even exclusive open could be listed as a capability as well.

Reply:
OK, I'll try to understand what is the desired solution to this and the "is_owner" comment, and write a follow-on patch next week.